### DISCUSSION TITLE
Elsa 3.1.3 publish event

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5318] [Q&A]
Please provide sample workflow with publish event

**Comment by glime-ai** [comment]
To provide you with a sample workflow that includes a publish event in Elsa, I will describe a simple two-workflow scenario where one workflow publishes an event and the other workflow is triggered by that event.

### Workflow 1: Event Publisher Workflow
1. **Start** activity: This is the entry point of the workflow.
2. **Publish Event** activity: This activity publishes an event. You need to specify the event name and optionally provide event data.
3. **WriteLine** activity: This activity logs a message indicating that the event has been published.

### Workflow 2: Event Subscriber Workflow
1. **Event** activity: This activity is configured to listen for the event published by Workflow 1. It acts as a trigger for this workflow.
2. **WriteLine** activity: This activity logs a message indicating that the event has been received and the workflow has been triggered.

Here is a JSON representation of the two workflows (simplified for readability):

```json
// Workflow 1: Event Publisher Workflow
{
  "activities": [
    {
      "type": "Start",
      "id": "start1"
    },
    {
      "type": "PublishEvent",
      "id": "publishEvent1",
      "eventName": "MyCustomEvent",
      "eventData": "Hello, World!"
    },
    {
      "type": "WriteLine",
      "id": "writeLine1",
      "text": "Event published."
    }
  ],
  "connections": [
    {
      "source": "start1",
      "target": "publishEvent1"
    },
    {
      "source": "publishEvent1",
      "target": "writeLine1"
    }
  ]
}

// Workflow 2: Event Subscriber Workflow
{
  "activities": [
    {
      "type": "Event",
      "id": "event1",
      "eventName": "MyCustomEvent"
    },
    {
      "type": "WriteLine",
      "id": "writeLine2",
      "text": "Event received."
    }
  ],
  "connections": [
    {
      "source": "event1",
      "target": "writeLine2"
    }
  ]
}
```

To implement these workflows in Elsa, you would use the Elsa Designer or write the workflows programmatically using C#. The JSON above is a conceptual representation and may not be directly importable into Elsa without additional context and configuration.

Please note that the actual implementation details, such as activity property names and values, may vary depending on the version of Elsa you are using and the specific requirements of your workflows.

If you need further assistance with creating these workflows in Elsa, please let me know which version of Elsa you are using and whether you prefer to use the designer or write the workflows programmatically.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
